network-api/networkapi/wagtailpages/pagemodels/buyersguide/forms.py

```diff
@@ -0,0 +1,35 @@
+from wagtail.admin.forms import WagtailAdminModelForm, WagtailAdminPageForm
+
+
+class BuyersGuideProductCategoryForm(WagtailAdminModelForm):
+    def clean_name(self):
+        instance = self.instance
+        set = instance.__class__.objects
+        name = self.cleaned_data["name"]
+        duplicate = set.filter(name__iexact=name).exclude(pk=instance.pk)
+
+        if hasattr(instance, "locale"):
+            duplicate = duplicate.filter(locale=instance.locale)
+
+        if duplicate.exists():
+            self.add_error("name", "A category with this name already exists.")
+
+        return name
+
+    def clean_parent(self):
+        parent = self.cleaned_data["parent"]
+        if parent:
+            if parent.parent:
+                self.add_error("parent", "Categories can only be two levels deep.")
+            if self.instance == parent:
+                self.add_error("parent", "A category cannot be a parent of itself.")
+        return parent
+
+
+class BuyersGuideArticlePageForm(WagtailAdminPageForm):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # Overriding admin form in order to make promote tab's
+        # "search image" and "search description" fields required.
+        self.fields["search_description"].required = True
+        self.fields["search_image"].required = True
```

Comment:
I also wonder if we maybe should do something closer to this section in the docs: https://docs.wagtail.org/en/stable/advanced_topics/customisation/page_editing_interface.html#custom-edit-handler-forms

... I think I am just realizing what the issue might be... 

We are overriding props of the form instance in our class, but our class does not have these fields. The super class also does not have these fields. The child class that is dynamically generated from our "base form class" is the one that has these fields. 

Maybe we should rather add the fields directly to the form class in the configuration we want them to be. They won't be overridden according to the docs. 

Or we should perform the checks in the `clean` method. This would have the downside that the UI does not show the fields as required.